src/v/cluster/tests/producer_state_tests.cc

```diff
@@ -346,3 +346,81 @@ FIXTURE_TEST(test_state_management_with_multiple_namespaces, test_fixture) {
         vp.producer->shutdown_input();
     }
 }
+
+FIXTURE_TEST(test_sequence_overflow_handling, test_fixture) {
+    create_producer_state_manager(10, 10);
+
+    auto producer = new_producer([&](model::producer_identity) {});
+    std::vector<producer_ptr> producers;
+    producers.push_back(producer);
+    auto defer = ss::defer([&] { clean(producers); });
+    auto calculate_record_count = [](seq_t first, seq_t last) {
+        if (last >= first) {
+            return last - first + 1;
+        } else {
+            // handle overflow
+            return (std::numeric_limits<seq_t>::max() - first + 1) + (last + 1);
+        }
+    };
+    auto make_bid = [&](seq_t first, seq_t last) {
+        return model::batch_identity{
+          .pid = producer->id(),
+          .first_seq = first,
+          .last_seq = last,
+          .record_count = calculate_record_count(first, last),
+          .max_timestamp = model::timestamp::now(),
+          .is_transactional = false};
+    };
+
+    auto result = producer->try_emplace_request(
+      make_bid(
+        std::numeric_limits<seq_t>::max() - 20,
+        std::numeric_limits<seq_t>::max() - 19),
+      model::term_id(1),
+      true);
+    BOOST_REQUIRE(!result.has_error());
+    // check if sequence validation fails correctly
+    auto result_expected_fail = producer->try_emplace_request(
+      make_bid(
+        std::numeric_limits<seq_t>::max() - 15,
+        std::numeric_limits<seq_t>::max() - 10),
+      model::term_id(1));
+    //
```

Comment:
nit: Formatting is a bit confusing to me. Could you maybe make each case (starting with a `try_emplace_request` line and ending with checks) run without blank lines, and separate cases with blank lines?